src/hints/tests.rs

```diff
@@ -0,0 +1,67 @@
+#[cfg(test)]
+mod tests {
+
+    use cairo_vm::serde::deserialize_program::ApTracking;
+    use cairo_vm::types::exec_scope::ExecutionScopes;
+    use num_bigint::BigInt;
+
+    use crate::hints::*;
+
+    macro_rules! references {
+        ($num:expr) => {{
+            let mut references = cairo_vm::stdlib::collections::HashMap::<usize, HintReference>::new();
+            for i in 0..$num {
+                references.insert(i as usize, HintReference::new_simple((i as i32 - $num)));
+            }
+            references
+        }};
+    }
+
+    macro_rules! ids_data {
+        ( $( $name: expr ),* ) => {
+            {
+                let ids_names = vec![$( $name ),*];
+                let references = references!(ids_names.len() as i32);
+                let mut ids_data = cairo_vm::stdlib::collections::HashMap::<cairo_vm::stdlib::string::String, HintReference>::new();
+                for (i, name) in ids_names.iter().enumerate() {
+                    ids_data.insert(cairo_vm::stdlib::string::ToString::to_string(name), references.get(&i).unwrap().clone());
+                }
+                ids_data
+            }
+        };
+    }
+
+    #[test]
+    fn test_is_on_curve() {
+        let mut vm = VirtualMachine::new(false);
+        vm.set_fp(1);
+        vm.add_memory_segment();
+        vm.add_memory_segment();
+
+        let ids_data = ids_data!["is_on_curve"];
+        let ap_tracking = ApTracking::default();
+
+        let mut exec_scopes = ExecutionScopes::new();
+
+        let y = BigInt::from(1234);
+        let y_square_int = y.clone() * y.clone();
+
+        exec_scopes.insert_value("y", y);
+        exec_scopes.insert_value("y_square_int", y_square_int);
```

Comment:
Should use `vars` mod as done in #4 (maybe merge that one first to avoid conflicts...)

Reply:
f9bc365